hoss/dimension_utilities.py

```diff
@@ -101,10 +101,214 @@ def prefetch_dimension_variables(
 
     # Create bounds variables if necessary.
     add_bounds_variables(required_dimensions_nc4, required_dimensions, varinfo, logger)
-
     return required_dimensions_nc4
 
 
+def get_override_projected_dimensions(
+    varinfo: VarInfoFromDmr,
+    override_variable_name: str,
+) -> str | None:
+    """returns the x-y projection variable names that would
+    match the geo coordinate names. The `latitude` coordinate
+    variable name gets converted to 'projected_y' dimension scale
+    and the `longitude` coordinate variable name gets converted to
+    'projected_x'
+
+    """
+    override_variable = varinfo.get_variable(override_variable_name)
+    if override_variable is not None:
+        if override_variable.is_latitude():
+            projected_dimension_name = 'projected_y'
+        elif override_variable.is_longitude():
+            projected_dimension_name = 'projected_x'
+        else:
+            projected_dimension_name = None
+    return projected_dimension_name
+
+
+def get_coordinate_variables(
+    varinfo: VarInfoFromDmr,
+    requested_variables: Set[str],
+) -> list[str]:
+    """This method returns coordinate variables that are referenced
+    in the variables requested. It returns it in a specific order
+    [latitude, longitude]
+    """
+
+    # try:
+    coordinate_variables_set = varinfo.get_references_for_attribute(
+        requested_variables, 'coordinates'
+    )
+    coordinate_variables = []
+    for coordinate in coordinate_variables_set:
+        if varinfo.get_variable(coordinate).is_latitude():
+            coordinate_variables.insert(0, coordinate)
+        elif varinfo.get_variable(coordinate).is_longitude():
+            coordinate_variables.insert(1, coordinate)
+
```

Comment:
I have a few thoughts here, but the most important one:

I think the code _needs_ to find both a longitude and a latitude to be able to work later on, so there should probably be an exception if you only find one of them.